xarray/structure/merge.py

```diff
@@ -793,18 +794,99 @@ def merge_core(
     return _MergeResult(variables, coord_names, dims, out_indexes, attrs)
 
 
+def merge_trees(
+    trees: Iterable[DataTree],
+    compat: CompatOptions | CombineKwargDefault = _COMPAT_DEFAULT,
+    join: JoinOptions | CombineKwargDefault = _JOIN_DEFAULT,
+    fill_value: object = dtypes.NA,
+    combine_attrs: CombineAttrsOptions = "override",
+) -> DataTree:
+    """Merge specialized to DataTree objects."""
+    from xarray.core.dataset import Dataset
+    from xarray.core.datatree import DataTree
+    from xarray.core.datatree_mapping import add_path_context_to_errors
+
+    if fill_value is not dtypes.NA:
+        # fill_value support dicts, which probably should be mapped to sub-groups?
+        raise NotImplementedError(
+            "fill_value is not yet supported for DataTree objects in merge"
+        )
+
+    node_lists: defaultdict[str, list[DataTree]] = defaultdict(list)
+    for tree in trees:
+        for key, node in tree.subtree_with_keys:
+            node_lists[key].append(node)
+
+    root_datasets = [node.dataset for node in node_lists.pop(".")]
+    with add_path_context_to_errors("."):
+        root_ds = merge(
+            root_datasets, compat=compat, join=join, combine_attrs=combine_attrs
+        )
+    result = DataTree(dataset=root_ds)
+
+    def depth(kv):
+        return kv[0].count("/")
```

Comment:
Maybe you could use https://github.com/pydata/xarray/blob/ad5140491f9d670f9229db1935250bf6c55aa246/xarray/core/treenode.py#L473

Reply:
`Tree.level` turns out to be the right property. `depth` refers (somewhat confusing, IMO) to the number of levels _below_ the given node.